src/main/java/com/google/api/codegen/transformer/JavaTestTransformer.java

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.ApiConfig;
+import com.google.api.codegen.InterfaceView;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.viewmodel.ViewModelDoc;
+import com.google.api.codegen.viewmodel.testing.ApiTestClassView;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class JavaTestTransformer implements Transformer {
```

Comment:
I renamed the main surface transformer to JavaGapicSurfaceTransformer; I think a good name for this class is JavaGapicSurfaceTestTransformer.